packages/adapter-better-sqlite3/src/conversion.ts

```diff
@@ -150,12 +150,8 @@ export function mapRow(row: Row, columnTypes: ColumnType[]): ResultValue[] {
   for (let i = 0; i < row.length; i++) {
     const value = row[i]
 
-    // Convert array buffers to arrays of bytes.
-    // Base64 would've been more efficient but would collide with the existing
-    // logic that treats string values of type Bytes as raw UTF-8 bytes that was
-    // implemented for other adapters.
-    if (value instanceof ArrayBuffer || value instanceof Buffer) {
-      result[i] = Array.from(new Uint8Array(value))
+    if (value instanceof ArrayBuffer) {
+      result[i] = new Uint8Array(value)
```

Comment:
`better-sqlite3` returns `Buffer` for `BLOB` affinity according to https://github.com/WiseLibs/better-sqlite3/blob/master/docs/api.md#binding-parameters.

`Buffer` is a subclass of `Uint8Array`.
`ArrayBuffer` is not an ancestor of `Uint8Array`.